src/libstd/io/stdio.rs

```diff
@@ -872,6 +874,83 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that a prompt
+/// is printed before capturing input. This function returns
+/// either success ([`Ok`]) or failure ([`Err`]) containing
+/// the input. See the [`io::Result`] module documentation
+/// for details.
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(prompt)]
+/// use std::io::prompt;
+///
+/// fn main() {
+///     let name = prompt("Enter name: ").expect("input failed!");
+///
+///     println!("Your name is {}!", name);
+/// }
+/// ```
+#[unstable(
+    feature = "prompt",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn prompt(prompt: &str) -> Result<String> {
+    stdout().write(prompt.as_bytes())?;
+    input()
+}
+
+/// Constructs a new handle to the standard input of the current
+/// process, locks this handle and reads a line of input, returning
+/// either success ([`Ok`]) or failure ([`Err`]) containing the input.
+/// See the [`io::Result`] module documentation for details. For
+/// automatic prompt handling, see the [`prompt`] method.
```

Comment:
I don't think handles and locking are very relevant to this function and this is likely to be confusing to new users who will be using this function. Consider significantly simplifying the wording in the documentation.

Here is the documentation for the Python `input` as a comparison:
```
input(prompt=None, /)
    Read a string from standard input.  The trailing newline is stripped.
    
    The prompt string, if given, is printed to standard output without a
    trailing newline before reading input.
    
    If the user hits EOF (*nix: Ctrl-D, Windows: Ctrl-Z+Return), raise EOFError.
    On *nix systems, readline is used if available.
```

Reply:
I agree, the docs should describe the observable properties of a function (i.e. what matters to the caller) and (generally) not implementation details.

Reply:
fixed function explanation.